### PR TITLE
test(oas-utils): more tests for `importSpecToWorkspace`

### PR DESCRIPTION
**Pull request by hanspagel**
This PR improve a few code comments and adds a bunch of tests for `importSpecToWorkspace`.

It’s also adding a console.log for the time it takes to create the workspace, and removes the console.log for the time it takes to create the search index. The search index was never really blocking anything, and it was rather quick, even for big files. The workspace becomes more relevant for the time to interaction, so I’d like to keep an eye on it for real-world examples. 

The diff is hard to follow, but no actual code is touched (except the mentioned logging).

### CODE REVIEW

File: packages/api-client/src/store/import-spec.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
   importSpecToWorkspace,
 } from '@scalar/oas-utils/transforms'
 import type { OpenAPIV3, OpenAPIV3_1 } from '@scalar/openapi-types'
-import type { ReferenceConfiguration, Spec } from '@scalar/types/legacy'
```
Comment (hanspagel) [inline packages/api-client/src/store/import-spec.ts:9]: unused import

File: packages/oas-utils/src/transforms/import-spec.test.ts [hunk 2]
```diff
@@ -121,9 +185,111 @@ describe('Import OAS Specs', () => {
       const authScheme = res.securitySchemes[5] as SecuritySchemeOauth2
       expect(authScheme['x-scalar-client-id']).toEqual(testId)
     })
+
+    it('handles empty security requirements', async () => {
+      const specWithEmptySecurity = {
+        ...galaxy,
+        security: [{}],
```
Comment (amritk) [inline packages/oas-utils/src/transforms/import-spec.test.ts:192]: this is not an empty requirement btw, it is an optional one 😉 `[]` would be empty

Comment (amritk) [reply packages/oas-utils/src/transforms/import-spec.test.ts:192]: not that it matters

**Review by amritk — APPROVED** [review]
More tests are always welcome!

**Review by marclave — APPROVED** [review]
